### PR TITLE
netcoredbg: 2.2.0-961 -> 3.0.0-1018

### PR DESCRIPTION
**Pull request by konradmalik**
## Description of changes

- updated netcoredbg to 3.0.0-1018 (https://github.com/Samsung/netcoredbg/releases)
- removed patches that are no longer needed
- removed update script (it was never used, if something better can be done here pls let me know)
- updated dotnet to 7.0.14
- updated deps.nix
- added a simple version test

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by leo60228** [comment]
The build fails for me:

```
[ 23%] Compiling generrmsg/bin/generrmsg.dll
MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
/nix/store/gm3ak3vgrx8qik40cnw28056i38hajsb-dotnet-sdk-8.0.100/sdk/8.0.100/NuGet.targets(156,5): error : Unable to load the service index for source /api.nuget.org/v3/index.json. [/build/source/build/src/generrmsg/generrmsg.csproj]
/nix/store/gm3ak3vgrx8qik40cnw28056i38hajsb-dotnet-sdk-8.0.100/sdk/8.0.100/NuGet.targets(156,5): error :   Resource temporarily unavailable (api.nuget.org:443) [/build/source/build/src/generrmsg/generrmsg.csproj]
/nix/store/gm3ak3vgrx8qik40cnw28056i38hajsb-dotnet-sdk-8.0.100/sdk/8.0.100/NuGet.targets(156,5): error :   Resource temporarily unavailable [/build/source/build/src/generrmsg/generrmsg.csproj]

Build FAILED.
```

**Comment by konradmalik** [comment]
yeah, not sure what's wrong but it's connected with dotnet 8. Seems like only x86_64-darwin works with 8. Maybe it's missing some entries in deps.nix and tries to get them from nuget directly? Not sure.

I rolled back the dotnet upgrade so we're again using dotnet 7. Netcoredbg build instructions still refer to dotnet 7 so I think we'll just migrate to 8 once the upstream repo migrates to it.

I've tested and it builds and runs on x86_64-darwin and aarch64-linux now. We'll see what happens in checks.

File: pkgs/development/tools/misc/netcoredbg/default.nix [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
   };
 
   dotnet-sdk = dotnetCorePackages.sdk_7_0;
 
   src = fetchFromGitHub {
+    inherit sha256;
     owner = "Samsung";
     repo = pname;
     rev = version;
-    sha256 = hash;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/netcoredbg/default.nix:21]:
```suggestion
    owner = "Samsung";
    repo = pname;
    rev = version;
    inherit hash;
```

File: pkgs/development/tools/misc/netcoredbg/default.nix [hunk 2]
```diff
@@ -1,34 +1,31 @@
-{ lib, clangStdenv, stdenv, cmake, autoPatchelfHook, fetchFromGitHub, dotnetCorePackages, buildDotnetModule }:
+{ lib, clangStdenv, stdenv, cmake, autoPatchelfHook, fetchFromGitHub, dotnetCorePackages, buildDotnetModule, netcoredbg, testers }:
 let
   pname = "netcoredbg";
-  version = "2.2.0-961";
-  hash = "0gbjm8x40hzf787kccfxqb2wdgfks81f6hzr6rrmid42s4bfs5w7";
+  build = "1018";
+  release = "3.0.0";
+  version = "${release}-${build}";
+  sha256 = "sha256-9Rd0of1psQTVLaTOCPWsYgvIXN7apZKDQNxuZGj4vXw=";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/netcoredbg/default.nix:7]:
```suggestion
  hash = "sha256-9Rd0of1psQTVLaTOCPWsYgvIXN7apZKDQNxuZGj4vXw=";
```

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

**Comment by konradmalik** [comment]
> Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) when naming your commits.

hope it's better now. I've renamed the PR as well. Please let me know if I should split this into separate commits (eg. a separate one for the test).

File: pkgs/development/tools/misc/netcoredbg/default.nix (hunk 1 above)
Comment (konradmalik) [reply pkgs/development/tools/misc/netcoredbg/default.nix]: please see the newest version

File: pkgs/development/tools/misc/netcoredbg/default.nix (hunk 2 above)
Comment (konradmalik) [reply pkgs/development/tools/misc/netcoredbg/default.nix]: please see the newest version

**Review by leo60228 — APPROVED** [review]
LGTM, tested and working

**Review by mdarocha — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Pull request merged**